### ISSUE TITLE
MasterPathHandlers::HandleGetUnderReplicationStatus may crash dereferencing the Result

### ISSUE THREAD

**Issue by tedyu** [yugabyte/yugabyte-db#6662]
From https://detective-gcp.dev.yugabyte.com/job/github-yugabyte-db-phabricator%2F68889%2Fartifact%2Fbuild%2Ftsan-clang-dynamic-ninja%2Fyb-test-logs%2Ftests-integration-tests__master_path_handlers-itest%2FMasterPathHandlersItest_TestTabletUnderReplicationEndpoint.log?class=MasterPathHandlersItest&max_lines=3000&name=TestTabletUnderReplicationEndpoint&start_line=6001

```
@ 0x7f756176ac61 yb::(anonymous namespace)::DumpStackTraceAndExit() (src/yb/util/logging.cc:166)
--
6352 | @ 0x7f7560b88142
6353 | @ 0x7f7560b88c9c
6354 | @ 0x7f7560b8d07a
6355 | @ 0x7f756fbf09bf yb::Result<std::vector<scoped_refptr<yb::master::TabletInfo>, std::allocator<scoped_refptr<yb::master::TabletInfo> > > >::get_ptr() (src/yb/util/result.h:0)
6356 | @ 0x7f756fbefaf9 yb::Result<std::vector<scoped_refptr<yb::master::TabletInfo>, std::allocator<scoped_refptr<yb::master::TabletInfo> > > >::operator*() & (src/yb/util/result.h:196)
6357 | @ 0x7f756fbe97b7 yb::master::MasterPathHandlers::HandleGetUnderReplicationStatus(yb::WebCallbackRegistry::WebRequest const&, yb::WebCallbackRegistry::WebResponse*) (src/yb/master/master-path-handlers.cc:1300)
6358 | @ 0x7f756fbf5d5c decltype(*(std::forward<yb::master::MasterPathHandlers*&>(fp0)).*fp(std::forward<yb::WebCallbackRegistry::WebRequest const&>(fp1), std::forward<yb::WebCallbackRegistry::WebResponse*>(fp1))) std::__invoke<void (yb::master::MasterPathHandlers::*&)(yb::WebCallbackRegistry::WebRequest const&, yb::WebCallbackRegistry::WebResponse*), yb::master::MasterPathHandlers*&, yb::WebCallbackRegistry::WebRequest const&, yb::WebCallbackRegistry::WebResponse*, void>(void (yb::master::MasterPathHandlers::*&)(yb::WebCallbackRegistry::WebRequest const&, yb::WebCallbackRegistry::WebResponse*), yb::master::MasterPathHandlers*&, yb::WebCallbackRegistry::WebRequest const&, yb::WebCallbackRegistry::WebResponse*&&) (/opt/yb-build/thirdparty/yugabyte-db-thirdparty-v20201111062958-7047513803-centos7-linuxbrew/installed/tsan/libcxx/include/c++/v1/type_traits:4286)
```
The crash was in the CHECK:
```
  auto get_ptr() {
#ifndef NDEBUG
    CHECK(ANNOTATE_UNPROTECTED_READ(success_checked_));
```

**Comment by tedyu** [comment]
```
diff --git a/src/yb/master/master-path-handlers.cc b/src/yb/master/master-path-handlers.cc
index 343d9a944..234655286 100644
--- a/src/yb/master/master-path-handlers.cc
+++ b/src/yb/master/master-path-handlers.cc
@@ -1297,13 +1297,15 @@ void MasterPathHandlers::HandleGetUnderReplicationStatus(const Webserver::WebReq
   jw.String("underreplicated_tablets");
   jw.StartArray();

-  for(TabletInfoPtr t : *underreplicated_ts) {
-    jw.StartObject();
-    jw.String("table_uuid");
-    jw.String(t->table()->id());
-    jw.String("tablet_uuid");
-    jw.String(t.get()->tablet_id());
-    jw.EndObject();
+  if (underreplicated_ts.ok()) {
+    for(TabletInfoPtr t : *underreplicated_ts) {
+      jw.StartObject();
+      jw.String("table_uuid");
+      jw.String(t->table()->id());
+      jw.String("tablet_uuid");
+      jw.String(t.get()->tablet_id());
+      jw.EndObject();
+    }
   }

   jw.EndArray();
```

**Comment by sanketkedia** [comment]
Closed by commit: [f3cd833](https://github.com/yugabyte/yugabyte-db/commit/f3cd83374f24b2444ac40030dc5fedc7743ee206)

Phabricator diff: [10031](https://phabricator.dev.yugabyte.com/D10031)
